pysatSeasons/avg.py

```diff
@@ -1,372 +1,459 @@
-    if len(objidx) > 0:
-        for zk in zarr[objidx]:
-            medianAns[zk] = np.zeros((numy, numx)) * np.nan
-            countAns[zk] = np.zeros((numy, numx)) * np.nan
-            devAns[zk] = np.zeros((numy, numx)) * np.nan
-            for yj in yarr:
-                for xi in xarr:
-                    # convert deque storing data into numpy array
-                    ans[zk][yj][xi] = np.array(ans[zk][yj][xi])
-                    # filter out an NaNs in the arrays
-                    idx, = np.where(np.isfinite(ans[zk][yj][xi]))
-                    ans[zk][yj][xi] = (ans[zk][yj][xi])[idx]
-                    # perform median averaging
-                    if len(idx) > 0:
-                        medianAns[zk][yj, xi] = np.median(ans[zk][yj][xi])
-                        countAns[zk][yj, xi] = len(ans[zk][yj][xi])
-                        devAns[zk][yj, xi] = np.median(abs(ans[zk][yj][xi]
-                                                       - medianAns[zk][yj, xi]))
-
-    # prepare output
+                    if len(temp_median[yj][xi]) > 0:
+                        key = [name for name in temp_median[yj][xi].data_vars]
+                        median_ans[zk][yj, xi] = temp_median[yj][xi][key[0]]
+                        count_ans[zk][yj, xi] = temp_count[yj][xi]
+                        dev_ans[zk][yj, xi] = temp_dev[yj][xi][key[0]]
+
+    # Prepare output
     output = {}
     for i, label in enumerate(data_label):
-        output[label] = {'median': medianAns[i],
-                         'count': countAns[i],
-                         'avg_abs_dev': devAns[i],
+        output[label] = {'median': median_ans[i],
+                         'count': count_ans[i],
+                         'avg_abs_dev': dev_ans[i],
                          'bin_x': binx,
                          'bin_y': biny}
 
-        if returnData:
+        if return_data:
             output[label]['data'] = ans[i]
 
     return output
 
 
-# simple averaging through multiple iterations
-
 def mean_by_day(inst, data_label):
-    """Mean of data_label by day over Instrument.bounds
+    """Mean of `data_label` by day over `Instrument.bounds`.
 
     Parameters
     ----------
-    data_label : string
-        string identifying data product to be averaged
+    inst : pysat.Instrument
+        Instrument object to perform mean upon.
+    data_label : str
+        Data product label to be averaged.
 
     Returns
     -------
-    mean : pandas Series
-        simple mean of data_label indexed by day
+    mean : pandas.Series
+        Mean of `data_label` indexed by day.
+
+    Note
+    ----
+    The range of dates to be loaded, and the cadence used to load data over
+    that range, is controlled by the `inst.bounds` attribute.
 
     """
     return _core_mean(inst, data_label, by_day=True)
 
 
 def mean_by_orbit(inst, data_label):
-    """Mean of data_label by orbit over Instrument.bounds
+    """Mean of `data_label` by orbit over Instrument.bounds.
```

Comment:
For these 'mean' functions, are you calculating the mean?  If so, recommend using the word 'Calculate' in the docstrings.